src/pages/iou/request/step/IOURequestStepSplitPayer.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useMemo} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {usePersonalDetails} from '@components/OnyxProvider';
+import SelectionList from '@components/SelectionList';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useLocalize from '@hooks/useLocalize';
+import useScreenWrapperTranstionStatus from '@hooks/useScreenWrapperTransitionStatus';
+import * as IOUUtils from '@libs/IOUUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import type {OptionData} from '@libs/ReportUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import type {Participant} from '@src/types/onyx/IOU';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import type {WithWritableReportOrNotFoundProps} from './withWritableReportOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+type IOURequestStepSplitPayerProps = WithWritableReportOrNotFoundProps<typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT> & {
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
+};
+
+function IOURequestStepSplitPayer({
+    route: {
+        params: {iouType, transactionID, action, backTo},
+    },
+    transaction,
+    report,
+}: IOURequestStepSplitPayerProps) {
+    const {translate} = useLocalize();
+    const personalDetails = usePersonalDetails();
+    const {didScreenTransitionEnd} = useScreenWrapperTranstionStatus();
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+
+    const currentUserOption = useMemo(
+        () => ({
+            accountID: currentUserPersonalDetails.accountID,
+            searchText: currentUserPersonalDetails.login,
+            selected: true,
+        }),
+        [currentUserPersonalDetails],
+    );
+
+    const sections = useMemo(() => {
+        const participants = transaction?.participants ?? [];
+        const participantOptions =
+            [currentUserOption, ...participants]
+                ?.filter((participant) => Boolean(participant.accountID))
+                ?.map((participant) => {
+                    const participantAccountID = participant.accountID ?? 0;
+                    return participantAccountID ? OptionsListUtils.getParticipantsOption(participant, personalDetails) : OptionsListUtils.getReportOption(participant);
```

Comment:
Participants in this page are never reports, so we can return participant options directly

Reply:
Updated.